new packages: JACK Audio Connection Kit and `alsa-lib`

knyipab opened the pull request:
Few packages
* `jack2` is an audio server. On Android, it can use OpenSLES as backend. 
* The metapackage `jack` is for dependency management and will accept the upcoming `pipewire-jack` as an option. By the way, I am not interested in making `jack1` functional and be one of the `jack` option, given that we have `jack2` now and it seems Linux distros usually not packing `jack1` officially (at least for Archlinux it is in AUR). 
* `alsa-lib` as deps of `jack2`. Also, it is tested to work along with `pipewire-alsa` which will be finalized in the pipewire PR. 
* `mpv` and `mpv-x` are updated to support jack and alsa audio output

Tested working with: 
```
jackd -d opensles
```
and then use the updated mpv in this PR
```
mpv --ao=jack ./test.wav
```

By the way, since https://github.com/termux/termux-packages/pull/20353 also use jack (which I put a stub and useless jack1), I will refactor that PR after this is merged.

Grimler91 commented on packages/mpv/build.sh:
```diff
@@ -4,10 +4,12 @@ TERMUX_PKG_LICENSE="GPL-2.0"
 TERMUX_PKG_AUTO_UPDATE=false
-TERMUX_PKG_DEPENDS="ffmpeg, libandroid-glob, libandroid-support, libarchive, libass, libcaca, libiconv, liblua52, libsixel, libuchardet, openal-soft, pulseaudio, rubberband, zlib, libplacebo"
+TERMUX_PKG_DEPENDS="alsa-lib, ffmpeg, jack, libandroid-glob, libandroid-support, libarchive, libass, libcaca, libiconv, liblua52, libsixel, libuchardet, openal-soft, pulseaudio, rubberband, zlib, libplacebo"
+TERMUX_PKG_BUILD_DPENEDS="jack2"
```
Typo, DPENEDS -> DEPENDS. 
It works anyways at the moment since package depends on jack which depends on jack2, I don't think there is any reason to add jack2 to BUILD_DEPENDS

knyipab replied:
Thanks for pointing out! And I removed that and just force-pushed. 

I thought when the metapackage `jack` indicates that (which I took reference to `vulkan-loader`): 
```
TERMUX_PKG_DEPENDS="jack2"
TERMUX_PKG_ANTI_BUILD_DEPENDS="jack2"
```
, then in build time `mpv` will not have `jack2` available. 

Instead, it seems that the first candidate of metapackage is installed during build time, right?

knyipab replied:
It is still so puzzling to me why `mpv` can detect jack2 backend in build time (the built `mpv --ao=help` shows jack). I am asking because `ardour` cannot detect jack backend when `jack2` is not on the `TERMUX_PKG_BUILD_DEPENDS` (https://github.com/knyipab/termux-packages/actions/runs/9529197300). 
And FYR that `dev/pipewire-group` branch has a slightly different `jack` metapackage: 
```
TERMUX_PKG_DEPENDS="pipewire-jack | jack2"
TERMUX_PKG_ANTI_BUILD_DEPENDS="pipewire-jack, jack2"
```

Would that be because `mpv` and `jack2` are in the same pull request. So the leftover of `jack2` build take effect on building `mpv`? That case should we keep `TERMUX_PKG_BUILD_DEPENDS="jack2"`?

Or should I remove `TERMUX_PKG_ANTI_BUILD_DEPENDS` which I copied from `vulkan-loader`? Is that `vulkan-loader` uses anti build deps because the that two libs can't be linked in the same way? while jack2 and pipewire-jack can be linked the same way.

Grimler91 commented on x11-packages/mpv-x/build.sh:
```diff
@@ -5,10 +5,12 @@ TERMUX_PKG_LICENSE="GPL-2.0"
 TERMUX_PKG_AUTO_UPDATE=false
-TERMUX_PKG_DEPENDS="ffmpeg, libandroid-glob, libandroid-shmem, libarchive, libass, libbluray, libcaca, libdrm, libdvdnav, libiconv, libjpeg-turbo, liblua52, libsixel, libuchardet, libx11, libxext, libxinerama, libxpresent, libxrandr, libxss, libzimg, littlecms, openal-soft, pipewire, pulseaudio, rubberband, zlib, libplacebo"
+TERMUX_PKG_DEPENDS="alsa-lib, ffmpeg, jack, libandroid-glob, libandroid-shmem, libarchive, libass, libbluray, libcaca, libdrm, libdvdnav, libiconv, libjpeg-turbo, liblua52, libsixel, libuchardet, libx11, libxext, libxinerama, libxpresent, libxrandr, libxss, libzimg, littlecms, openal-soft, pipewire, pulseaudio, rubberband, zlib, libplacebo"
+TERMUX_PKG_BUILD_DPENEDS="jack2"
```
Typo, DPENEDS -> DEPENDS (and maybe not necessary as above)

Grimler91 reviewed (changes requested):
Thanks for the PR, exciting work! I think we can drop BUILD_DEPENDS on jack2 for mpv, otherwise looks good!

knyipab commented:
Refactored to remove ANTI_BUILD_DEPENDS, added few utils pacakges and added 0004 patch to `jack2` to remove some superuser features, paving the way to debug with jack-enabled `ardour` in the future. 

There are quite some issues in `alsa-lib` and `jack2`, but still, I suggest merging this PR first because
1. Those issues are mostly beyond the scope of this PR
2.  `shotcut` (PR #20353) needs `jack` and `pipewire-alsa` (PR #20483) will need `alsa-lib`. 
3. There should be little to no harm to merge cuz existing packages aren't compiled with alsa-lib and jack. Fixes can be done later each time we enable alsa and jack features for other packages.

* ALSA has no access to kernel driver thru /dev/snd/*
  * rawmidi: implementation with Amidi NDK and MidiManager JDK will be a substantial work and probably need Termux:API
  * seq: a user-space alsa sequencer server probably helps https://github.com/hselasky/alsa-seq-server?tab=readme-ov-file
* `jack2` prompts quite some errors when using with a jack-enabled `ardour`

TomJo2000 commented on packages/alsa-utils/build.sh:
Since this reinstates support for `alsa-utils` I have removed it from disabled-packages in https://github.com/termux/termux-packages/commit/7a1f71a3c6400ee1b4cef7ab981cbacd917dd258.

Please check that `volume_mapping.c.patch` is not still applicable to the reinstated version.
https://github.com/termux/termux-packages/blob/2d69c057096807a0f09daee02e301d45081b6722/disabled-packages/alsa-utils/volume_mapping.c.patch#L1-L11

knyipab replied:
There is no #define exp10(x) and no use of exp10 in the current version of alsa-utils.
